### PR TITLE
Adding Admin DS ballerina DAO implementation

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Purpose
Fix for : https://github.com/wso2/apk/issues/272

### CODE REVIEW

File: admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    if dbClient is error {
+        return "Error while retrieving connection";
+    } else {
+        string org = "carbon.super";
+        sql:ParameterizedQuery query = `INSERT INTO APPLICATION_USAGE_PLAN (NAME, DISPLAY_NAME, 
+        ORGANIZATION, DESCRIPTION, QUOTA_TYPE, QUOTA, UNIT_TIME, TIME_UNIT, IS_DEPLOYED, UUID) 
+        VALUES (${atp.policyName},${atp.displayName},${org},${atp.description},${atp.defaultLimit.'type},
+        ${atp.defaultLimit.requestCount?.requestCount},${atp.defaultLimit.requestCount?.unitTime},
+        ${atp.defaultLimit.requestCount?.timeUnit},${atp.isDeployed},${atp.policyId})`;
+        io:println(query);
```
Comment (tharindu1st) [inline admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal:34]: lets make log:printdebug

File: admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+            io:println(result);
+            return "Error while inserting data into Database";  
+        }
+    }
+}
+
+public function getApplicationUsagePlanByIdDAO(string policyId) returns string?|ApplicationThrottlePolicy {
+    jdbc:Client | error dbClient  = getConnection();
+    if dbClient is error {
+        return "Error while retrieving connection";
```
Comment (tharindu1st) [inline admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal:48]: return Error type

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    } else {
+        string org = "carbon.super";
+        sql:ParameterizedQuery query = `INSERT INTO APPLICATION_USAGE_PLAN (NAME, DISPLAY_NAME, 
+        ORGANIZATION, DESCRIPTION, QUOTA_TYPE, QUOTA, UNIT_TIME, TIME_UNIT, IS_DEPLOYED, UUID) 
+        VALUES (${atp.policyName},${atp.displayName},${org},${atp.description},${atp.defaultLimit.'type},
+        ${atp.defaultLimit.requestCount?.requestCount},${atp.defaultLimit.requestCount?.unitTime},
+        ${atp.defaultLimit.requestCount?.timeUnit},${atp.isDeployed},${atp.policyId})`;
+        io:println(query);
+        sql:ExecutionResult | sql:Error result =  dbClient->execute(query);
+        if result is sql:ExecutionResult {
```
Comment (Krishanx92) [inline admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal:36]: Make sure to close the connection in all places

Comment (CrowleyRajapakse) [reply admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal:36]: fixed and resolved

File: admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal (hunk 2 above)
Comment (CrowleyRajapakse) [reply admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal]: fixed and resolved

File: admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal (hunk 1 above)
Comment (CrowleyRajapakse) [reply admin/admin-domain-service/ballerina/throttlingPolicyDAO.bal]: fixed and resolved

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**